common/libs/validators.ts

```diff
@@ -86,7 +86,13 @@ function validateEtherAddress(address: string): boolean {
 
 export function isValidPrivKey(privkey: string | Buffer): boolean {
   if (typeof privkey === 'string') {
-    return privkey.length === 64;
+    const initialCheck =
+      privkey.length === 64 && /^[a-fA-F0-9]+$/g.test(privkey);
+    if (initialCheck) {
+      const keyBuffer = Buffer.from(privkey, 'hex');
+      return isValidPrivate(keyBuffer);
+    }
+    return false;
   } else if (privkey instanceof Buffer) {
     return privkey.length === 32;
```

Comment:
Could you add the `isValidPrivate` check here too?
Should be able to do the check just once after the string is converted to a buffer.